Google.PowerShell/Container/GkeClusterCmdlets.cs

```diff
@@ -580,4 +581,381 @@ protected override void ProcessRecord()
+    ///   </code>
+    ///   <para>
+    ///   Creates a cluster named "my-cluster" in the default zone of the default project using config
+    ///   $nodeConfig and network "my-network".</para>
+    /// </example>
+    /// <example>
+    ///   <code>
+    ///   PS C:\> Add-GkeCluster -MachineType "n1-standard-4" `
+    ///                          -ClusterName "my-cluster" `
+    ///                          -Description "My new cluster" `
+    ///                          -Subnetwork "my-subnetwork"
+    ///   </code>
+    ///   <para>
+    ///   Creates a cluster named "my-cluster" with description "my new cluster" in the default zone of
+    ///   the default project using machine type "n1-standard-4" for each Google Compute Engine VMs
+    ///   in the cluster. The cluster will use the subnetwork "my-subnetwork".
+    ///   </para>
+    /// </example>
+    /// <example>
+    ///   <code>
+    ///   PS C:\> Add-GkeCluster -ImageType "GCI" `
+    ///                          -ClusterName "my-cluster" `
+    ///                          -Zone "us-central1-a" `
+    ///                          -MasterCredential (Get-Credential) `
+    ///                          -DisableMonitoringService `
+    ///                          -AdditionalZone "us-central1-f"
+    ///   </code>
+    ///   <para>
+    ///   Creates a cluster named "my-cluster" in zone "us-central1-a" of the default project.
+    ///   Asides from "us-central1-a", the cluster's nodes will also be found at zone "us-central1-f".
+    ///   The cluster will not have Google Monitoring Service enabled to write metrics.
+    ///   The master node of the cluster will have credential supplied by (Get-Credential).
+    ///   Each node of the cluster will be of type GCI.
+    ///   </para>
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/container-engine/reference/rest/v1/NodeConfig)">
+    /// [Node Configs]
+    /// </para>
+    /// <para type="link" uri="(https://kubernetes.io/docs/user-guide/labels/)">
+    /// [Kubernetes Labels]
+    /// </para>
+    /// <para type="link" uri="(https://cloud.google.com/compute/docs/instances/preemptible)">
+    /// [Preemptible VM instances]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsCommon.Add, "GkeCluster")]
+    public class AddGkeCluster : GkeNodeConfigCmdlet
+    {
+        private class ParameterSetNames
+        {
+            public const string ByNodeConfig = "ByNodeConfig";
+            public const string ByValues = "ByValues";
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// Size of the disk attached to each node in the cluster, specified in GB.
+        /// The smallest allowed disk size is 10GB.
+        /// The default disk size is 100GB.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        [ValidateRange(10, int.MaxValue)]
```

Comment:
I think there's a quota that is set for the project but we don't really know that. Same for other parameters.